include/aspect/introspection.h

```diff
@@ -293,6 +293,13 @@ namespace aspect
        */
       IndexSets index_sets;
 
+      enum AdvectionMethod
+      {
+        field = 0,
+        particles = 1
+      };
+      std::map<std::string,AdvectionMethod> advection_methods;
```

Comment:
Is there a reason you go with a lookup by name instead of using the index (and put it in a std::vector)? The advantage would be that you don't have to worry about the case that you can not find the entries. Also please add comments to this variable.